backend/src/db/courses.js

```diff
@@ -182,20 +179,30 @@ const reduceCoursesByDate = async (courses) => {
                     course_type_id: course.course_type_id,
                     course_type_name: course.course_type_name,
                     teacher: course.teacher,
-                    location: course.location[0].dataValues.location,
-                    address: course.location[0].dataValues.address,
-                    eventId: course.teachingSession[0].dataValues.eventId,
-                    startDate: course.teachingSession[0].dataValues.startDate,
-                    endDate: course.teachingSession[0].dataValues.endDate,
-                    price: price,
+                    location: course.location[0] ? course.location[0].dataValues.location : null,
+                    address: course.location[0] ? course.location[0].dataValues.address : null,
+                    teachingSession: course.teachingSession,
                     reservedCount: reservedCount.count,
                 };
             })
     );
     return await mappedCourses.reduce((obj, course) => {
-        const date = datefns.format(course.startDate, 'MM-DD-YYYY');
-        obj[date] = obj[date] || [];
-        obj[date].push(course);
+        for (let teachingSession of course.teachingSession) {
```

Comment:
I would definitely prefer forEach functional loop. and inside it call another function in charge of this transformation.
It would be safer and cleaner if you do it the functional way. FP is perfect for data transformation